Rehaul: displaying data (#959)
appsmithorg/appsmith-docs: website/docs/core-concepts/data-access-and-binding/displaying-data-read/README.md

### Display data from async JS function

Widgets have fields/properties where you can bind data or trigger actions.

**Sync fields** are properties that expect input or data. For example, for an Input widget, properties such as `Default Value`, `Max Characters`, `Regex`, and `Error Message` expect input and are sync fields.

**Async fields** are properties that can trigger an action or perform an operation. For example, the properties like `OnTextChanged` and `OnSubmit` of an input widget are async fields. You can use these properties to execute an [action](/reference/appsmith-framework/widget-actions), [Query](/core-concepts/data-access-and-binding/querying-a-database#running-a-query) or a function within a [JS object](/core-concepts/writing-code/javascript-editor-beta).

To display the response from an asynchronous JS function in a synchronous field, you need to retrieve it using the  `.data` property as shown below

```javascript
{{JSObjectName.functionName.data}}
```

 <VideoEmbed host="youtube" videoId="yn_8gs5w04g" title="Display response from async function in widget field" caption="Display response from async function in widget field"/> 

> ### Display data from sync JS function

Reviewer: ```suggestion
### Display data from JS function
```

To display the response from a synchronous JS function in a widget field, call the function inside the JS Object as shown below:

``` javascript
{{JSObjectName.functionName()}}
```

> 

Reviewer: Here **Update widgets programmatically**

> ## Transforming data

Reviewer: ```suggestion
## Transform data
```

> You can use Javascript to transform Query data when binding it to a property. For example, consider a Query that returns an array of objects as shown below:  

Reviewer: ```suggestion
Each widget property has a specific data type that it validates its value against. If the data type mismatches, it will throw an error. You can use JavaScript to transform the data when binding it to a property. For example, consider a query that returns an array of objects, as shown below:  
```

]
```

> Suppose you have to populate this data in a [select](/reference/widgets/select.md) widget. A select widget only accepts an Array\<label, value> data type in its option field, so directly binding the query data leads to an error due to incorrect data type.

Reviewer: ```suggestion
Suppose you want to display this data in a [Select](/reference/widgets/select.md) widget. A Select widget only accepts data as an Array in the `{ "label": "string", "value": "string" }` format in its **Options** property, so you must transform the data from the query to pass it in the required structure.
```

> To connect this data to a select widget, you have to transform the data in the [options property](/reference/widgets/select#widget-properties).

Reviewer: ```suggestion
```

> **Transformation Code**

Reviewer: ```suggestion
**Example:  The following example iterates over the query data and returns it in the format required for displaying data in the Select widget **
```

> The following example iterates over the query data and returns it in an `Array<label, value>` format:

Reviewer: ```suggestion

```

  });
}}
```

> ## Further reading

Reviewer: Think we have links to Table and List

* [Capture Data](/core-concepts/data-access-and-binding/capturing-data-write)
* [Creating Workflows](/core-concepts/writing-code/workflows)

> * [Using JavaScript Promises](/core-concepts/writing-code/javascript-promises)

Reviewer: ```suggestion
remove this
```
